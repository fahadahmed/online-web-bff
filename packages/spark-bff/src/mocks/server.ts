import { setupServer } from 'msw/node';
import a2pCustomerHandlers from 'datasources/a2p/customer/mocks/handlers';
import accessHandlers from 'datasources/user/access/mocks/handlers';
import accountSummaryHandlers from 'datasources/customer/accountsService/mocks/handlers';
import addressHandlers from 'datasources/location/address/mocks/handlers';
import airpointsHandlers from 'datasources/airpoints/mocks/handlers';
import availableNumbers from 'datasources/shop/availableNumbers/mocks/handlers';
import balanceServiceHandlers from 'datasources/finance/balanceService/mocks/handlers';
import billPaymentHandlers from 'datasources/finance/paymentService/billPayment/mocks/handlers';
import cartServicesHandlers from 'datasources/shop/cart/mocks/handlers';
import checkoutServicesHandlers from 'datasources/shop/checkout/mocks/handlers';
import connectionDiagnosticsHandlers from 'datasources/line/assure/connectionDiagnostics/mocks/handlers';
import connectionPromiseHandler from 'datasources/line/assure/connectionPromise/mocks/handlers';
import contentServiceHandlers from 'datasources/utility/content/mocks/handlers';
import digitalBillHandlers from 'datasources/finance/digitalBill/mocks/handlers';
import identityServiceHandlers from 'datasources/user/identityService/mocks/handlers';
import lineConfiguarationHandlers from 'datasources/line/lineConfiguration/mocks/handlers';
import lineDetailServiceHandlers from 'datasources/line/lineDetails/mocks/handlers';
import lineDetailsHandlers from 'datasources/line/lineSummary/mocks/handlers';
import lineSpendHistoryHandlers from 'datasources/line/spend/mocks/handlers';
import lineSubscriptionsService from 'datasources/line/lineSubscriptionsService/mocks/handlers';
import lineUsageDetailsMocks from 'datasources/line/usage/usageDetailsService/mocks';
import lineUsageHistoryMocks from 'datasources/line/usage/usageHistoryService/mocks';
import lineUsageMocks from 'datasources/line/usage/mocks/handlers';
import lineUsageTransactionMocks from 'datasources/line/usage/usageTransactionsService/mocks';
import networkSettingsHandlers from 'datasources/technology/networkSettingsService/mocks/handlers';
import orderHandlers from 'datasources/shop/order/mocks/handlers';
import paymentServiceHandlers from 'datasources/finance/paymentService/mocks/handlers';
import pointOfAddressHandlers from 'datasources/location/pointOfInterest/mocks/handlers';
import pointOfOutageHandlers from 'datasources/location/pointOfInterest/outagesPointOfInterest/mocks/handlers';
import preferenceHandlers from 'datasources/user/preference/mocks/handlers';
import prepaidBalanceServiceHandlers from 'datasources/finance/prepaidBalanceService/mocks/handlers';
import productDetailServiceHandlers from 'datasources/product/details/mocks/handlers';
import productFiltersServiceHandlers from 'datasources/product/filters/mocks/handlers';
import productOffersHandlers from 'datasources/product/offers/mocks/handlers';
import productPromotionsMPDServiceHandlers from 'datasources/product/promotions/mpd/mocks/handlers';
import profileServiceHandlers from 'datasources/marketing/profileService/mocks/handlers';
import resourceHandlers from 'datasources/utility/resource/mocks/handlers';
import roamingRatesHandlers from 'datasources/product/roamingRates/mocks/handlers';
import subscriptionActivationHandlers from 'datasources/line/subscriptionActivationService/mocks/handlers';
import topupHandlers from 'datasources/finance/paymentService/topup/mocks/handlers';
import transactionsListHandlers from 'datasources/line/transactionsService/mocks/handlers';
import troubleshootHandlers from 'datasources/line/assure/connectionTroubleshooter/mocks/handlers';
import userInfoHandlers from 'datasources/user/userInfo/mocks/handlers';
import walletServiceHandlers from 'datasources/finance/walletService/mocks/handlers';

const handlers = [
  ...a2pCustomerHandlers,
  ...accessHandlers,
  ...accountSummaryHandlers,
  ...addressHandlers,
  ...availableNumbers,
  ...airpointsHandlers,
  ...balanceServiceHandlers,
  ...billPaymentHandlers,
  ...cartServicesHandlers,
  ...checkoutServicesHandlers,
  ...connectionDiagnosticsHandlers,
  ...connectionPromiseHandler,
  ...contentServiceHandlers,
  ...digitalBillHandlers,
  ...identityServiceHandlers,
  ...lineConfiguarationHandlers,
  ...lineDetailServiceHandlers,
  ...lineDetailsHandlers,
  ...lineSpendHistoryHandlers,
  ...lineSubscriptionsService,
  ...lineUsageDetailsMocks,
  ...lineUsageHistoryMocks,
  ...lineUsageMocks,
  ...lineUsageTransactionMocks,
  ...networkSettingsHandlers,
  ...orderHandlers,
  ...paymentServiceHandlers,
  ...pointOfAddressHandlers,
  ...pointOfOutageHandlers,
  ...preferenceHandlers,
  ...prepaidBalanceServiceHandlers,
  ...productDetailServiceHandlers,
  ...productFiltersServiceHandlers,
  ...productOffersHandlers,
  ...productPromotionsMPDServiceHandlers,
  ...profileServiceHandlers,
  ...resourceHandlers,
  ...roamingRatesHandlers,
  ...subscriptionActivationHandlers,
  ...topupHandlers,
  ...transactionsListHandlers,
  ...troubleshootHandlers,
  ...userInfoHandlers,
  ...walletServiceHandlers,
];

const server = setupServer(...handlers);
export default server;
