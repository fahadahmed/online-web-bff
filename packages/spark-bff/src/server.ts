import { ApolloServer } from 'apollo-server-fastify';
import {
  ApolloServerPluginLandingPageDisabled,
  ApolloServerPluginLandingPageGraphQLPlayground,
} from 'apollo-server-core';
import { logger } from 'apolloLogger';
import context from 'context';
import dataSources from 'datasources';
import { ApolloLoggerPlugin, ApplicationLogEvent } from 'logging';
import schema from 'schema';
import { formatResponse } from 'utils/formatResponse';
import { isDevelopmentEnvironment } from 'utils/isDevelopmentEnvironment';

process.on('uncaughtException', (err) => {
  logger.error(new ApplicationLogEvent({ message: err.message, context: {} }));
  throw err;
});

const apolloServer = new ApolloServer({
  schema,
  dataSources,
  formatResponse,
  logger,
  context,
  introspection: isDevelopmentEnvironment(),
  plugins: [
    new ApolloLoggerPlugin(
      `BFF Server Started. BUILD_ID: ${process.env.BUILD_BUILDID}`,
    ),
    isDevelopmentEnvironment()
      ? ApolloServerPluginLandingPageGraphQLPlayground()
      : ApolloServerPluginLandingPageDisabled(),
  ],
});

export default apolloServer;
