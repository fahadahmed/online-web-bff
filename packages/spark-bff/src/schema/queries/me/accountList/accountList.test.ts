import { createTestClient, gql } from 'test/apolloTestUtils';
import { server } from 'test/server';
import {
  accountListHandler,
  guestAccountListHandler,
  unauthorizedAccountListHandler,
} from 'datasources/customer/accountsService/mocks/handlers';

beforeEach(() => {
  jest.useFakeTimers();
});

afterAll(() => {
  jest.clearAllTimers();
});

test('Get a simple list of accounts and related product information', async () => {
  server.use(accountListHandler);
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query AccountList {
        me {
          accountList {
            accountNumber
            accountId
            customerNumber
            customerSegment
            segment
            balanceManagement
            status
            firstName
            lastName
            inCollections
            products {
              lineNumber
              assetId
              offerId
              offerName
              productInstanceId
              ifpEligible
            }
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.me).toMatchSnapshot();
});

test('accountList query returns an empty list when unauthenticated', async () => {
  server.use(unauthorizedAccountListHandler);
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query AccountList {
        me {
          accountList {
            accountNumber
            accountId
            customerNumber
            customerSegment
            balanceManagement
            status
            firstName
            lastName
            inCollections
            products {
              lineNumber
              assetId
              offerId
              offerName
              productInstanceId
              ifpEligible
            }
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.me).toMatchSnapshot();
});

test('accountList query returns an empty list when guest', async () => {
  server.use(guestAccountListHandler);
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query AccountList {
        me {
          accountList {
            accountNumber
            accountId
            customerNumber
            customerSegment
            balanceManagement
            status
            firstName
            lastName
            inCollections
            products {
              lineNumber
              assetId
              offerId
              offerName
              productInstanceId
              ifpEligible
            }
          }
        }
      }
    `,
  });

  expect(res.errors).toBeUndefined();
  expect(res.data!.me).toMatchSnapshot();
});
