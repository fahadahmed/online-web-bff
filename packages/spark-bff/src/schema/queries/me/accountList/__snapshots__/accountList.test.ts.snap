// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get a simple list of accounts and related product information 1`] = `
Object {
  "accountList": Array [
    Object {
      "accountId": "1-16PQ-3214",
      "accountNumber": "180691324",
      "balanceManagement": "PREPAID",
      "customerNumber": "180691308",
      "customerSegment": "Consumer",
      "firstName": "SHEENAL",
      "inCollections": false,
      "lastName": "KUMAR",
      "products": Array [
        Object {
          "assetId": "1-2E3RV5SV",
          "ifpEligible": false,
          "lineNumber": "01660802083",
          "offerId": "misc020789",
          "offerName": "Broadband Equipment",
          "productInstanceId": "1-2E3GTV79",
        },
        Object {
          "assetId": "1-27X5NNAT",
          "ifpEligible": false,
          "lineNumber": "096240152",
          "offerId": "misc050385",
          "offerName": "Netflix â€“ 6mo offer",
          "productInstanceId": "1-27X5NNAT",
        },
        Object {
          "assetId": "1-2ASBRIIH",
          "ifpEligible": false,
          "lineNumber": "096240151",
          "offerId": "bvbundle050504",
          "offerName": "Unplan Fibre 100 with Landline 12 months",
          "productInstanceId": "1-28YFWN77",
        },
      ],
      "segment": "personal",
      "status": "Active",
    },
    Object {
      "accountId": "1-1SSSYIWO",
      "accountNumber": "702894573",
      "balanceManagement": "POSTPAID",
      "customerNumber": "702894476",
      "customerSegment": "Consumer",
      "firstName": "SAHIL",
      "inCollections": false,
      "lastName": "KUMAR",
      "products": Array [
        Object {
          "assetId": "1-2F7FAV99",
          "ifpEligible": true,
          "lineNumber": "02102947234",
          "offerId": "mbundle050290",
          "offerName": "Mobile $59.99 Rollover Plan",
          "productInstanceId": "1-2F5IR35I",
        },
      ],
      "segment": "personal",
      "status": "Active",
    },
    Object {
      "accountId": "2-3ZZZZABCD",
      "accountNumber": "369121518",
      "balanceManagement": "POSTPAID",
      "customerNumber": "123456789",
      "customerSegment": "Consumer",
      "firstName": "Ash",
      "inCollections": false,
      "lastName": "Ketchum",
      "products": Array [],
      "segment": "personal",
      "status": "Active",
    },
  ],
}
`;

exports[`accountList query returns an empty list when guest 1`] = `
Object {
  "accountList": Array [],
}
`;

exports[`accountList query returns an empty list when unauthenticated 1`] = `
Object {
  "accountList": Array [],
}
`;
